config/rbac/watcher_editor_role.yaml

```diff
@@ -0,0 +1,31 @@
+# permissions for end users to edit watchers.
+apiVersion: rbac.authorization.k8s.io/v1
```

Comment:
I assume this might due to the previous we are not using kubebuilder cli to generate watcher controller correctly so that these two roles are missing. If you check kyma and moduletemplate, those two resources all have these roles, and they are defined by `+kubebuilder:rbac`